docs/spec.md

```diff
@@ -273,10 +273,47 @@ be represented by the symbols `min` or `max`, respectively; the `exclusive`
 annotation is not applicable when the symbols `min` or `max` are specified.
 A range may not contain both `min` and `max`.
 
+All ranges have a type. The type of the range is the same as that of the minimum and/or maximum values specified
+in the range list. If both a minimum and maximum values are specified (i.e. `min` and `max` are not used), then both of 
+those values must be of the same type. (For example, `range::[1995-12-06T, 55.4]` mixes values of the timestamp and 
+number types, and therefore is not a valid range.)
+
+##### RANGE&lt;NUMBER&gt;
+
+A `number` range includes all values of any numeric type (i.e. `float`, `int`, and `decimal`) that fall within that range mathematically. 
+When testing a value for inclusion, the range bounds and the value that is being tested will all be converted to
+`decimal` values for the sake of comparison. `number` ranges do not include values of any other type.
+
+##### RANGE&lt;TIMESTAMP&gt;
+
+All [`timestamp` values](https://amzn.github.io/ion-docs/docs/spec.html#timestamp) represent an instant in time. 
+A `timestamp` with limited precision (for example, a year-only timestamp like `2007T`) maps to an instant by assuming 
+that all unspecified time unit fields are effectively zero (or one for the month and
+day units). `2007T` would map to the instant represented by `2007-01-01T00:00.000-00:00`. 
```

Comment:
From the Ion Spec that you linked:

> Timestamps represent a specific moment in time, always include a local offset, _and are capable of arbitrary precision._ (emphasis added)

The approach you are taking here ignores the fact that timestamps have arbitrary precision—essentially converting them all to have _infinite_ precision. That could lead to unintuitive results for users of Ion Schema.

For example, `range::[2021-01-01T, 2021-01-03T]` has inclusive endpoints, so users may expect that a timestamp such as `2021-01-03T12:00:00Z` would be within the range, but that value is actually outside the range.

Also, `range::[exclusive::2021-01-01T, 2021-01-03T]` looks like it should exclude all of `2021-01-01`, but by forcing timestamps to be compared at infinite precision, a timestamp such as `2021-01-01T00:00:00.0001Z` would be inside the range.

Have we considered any alternatives that consider a margin of error when comparing values of differing precisions?

Reply:
According to an offline discussion we all agreed to go with the decimal based ordering.